### DISCUSSION TITLE
what is oak?

### DISCUSSION THREAD

**Discussion by mindplay-dk** [denoland/deno#27446] [Q&A]
various Deno [tutorials](https://docs.deno.com/examples/create_react_tutorial/) and [templates](https://github.com/denoland/react-vite-ts-template) frequently use something called [oak](https://oakserver.org/).

I see on the home page that it provides middleware and a router, but Deno has a [built-in server](https://docs.deno.com/runtime/fundamentals/http_server/), and this article even says "Deno recommends using Oak for building web servers" - but middleware (to my mind, in the first place) doesn't need to be much more than `request => response` functions that you compose and bootstrap at launch.

the home page doesn't explain much, it just points to the generated API reference documentation - this being the only real documentation, it's difficult to know where to start, or how much you need to learn.

by the looks of it, this is similar to dozens of other existing, established frameworks?

is everyone using this for their new Deno projects? why? is it good? 🙂

**Comment by marvinhagemeister** [comment]
Deno comes with a built-in server, but not with a built-in router. Using a router is pretty much the first thing people will be looking for, hence the recommendation to go with a framework.

`oak` in particular is lightweight and a good starting point for new users exploring Deno. There are many other frameworks out there that you can use instead. `Hono` is another popular one, or you could use `express` or `koa` which are more tailored to Node user's as they rely on Node APIs under the hood.

**Reply by mindplay-dk** [reply]
Thanks, Marvin. 🙂

I guess it looks nice enough - it doesn't look significantly different from `hono`, `express` or `koa`, and now I guess all the available middleware for those needs to be rewritten (again) which just makes me wonder (again) why we keep starting over on this... 😌
